libwallet/src/swap/fsm/seller_swap.rs

```diff
@@ -1377,89 +1402,96 @@ where
 		_context: &Context,
 		tx_conf: &SwapTransactionsConfirmations,
 	) -> Result<StateProcessRespond, ErrorKind> {
-		match input {
-			Input::Check => {
-				// Be greedy, check the deadline for locking
-				post_refund_if_possible(self.node_client.clone(), swap, tx_conf)?;
-
-				// Just waiting
-				if let Some(conf) = tx_conf.secondary_redeem_conf {
-					if conf >= swap.secondary_confirmations {
-						// We are done
-						swap.add_journal_message(format!(
-							"{} redeem transaction has enough confirmations. Trade is complete",
-							swap.secondary_currency
-						));
-						return Ok(StateProcessRespond::new(StateId::SellerSwapComplete));
+		if let Input::Check = input {
+			// Be greedy, check the deadline for locking
+			post_refund_if_possible(self.node_client.clone(), swap, tx_conf)?;
+
+			// Just waiting
+			if let Some(conf) = tx_conf.secondary_redeem_conf {
+				let secondary_redeem_conf = match swap.secondary_currency.is_btc_family() {
+					true => conf >= swap.secondary_confirmations,
+					_ => conf > 0,
+				};
+				if secondary_redeem_conf {
+					//for eth, now we try to transfer ethers from interal wallet to buyers' eth wallet.
+					if !swap.secondary_currency.is_btc_family()
+						&& swap.eth_redirect_to_private_wallet
+					{
+						self.swap_api.transfer_scondary(swap)?;
 					}
+					// We are done
+					swap.add_journal_message(format!(
+						"{} redeem transaction has enough confirmations. Trade is complete",
+						swap.secondary_currency
+					));
+					return Ok(StateProcessRespond::new(StateId::SellerSwapComplete));
+				}
 
-					// If transaction was published for a while ago and still in mem pool. we need to bump the fees.
-					// It is applicable to BTC only
-					if swap.secondary_currency == Currency::Btc && conf == 0 {
-						match swap.posted_secondary_height {
-							Some(h) => {
-								if h < tx_conf.secondary_tip
-									- state::SECONDARY_HEIGHT_TO_INCREASE_FEE
+				// If transaction was published for a while ago and still in mem pool. we need to bump the fees.
+				// It is applicable to BTC only
+				if swap.secondary_currency == Currency::Btc && conf == 0 {
```

Comment:
Do we need the same for eth?  We don't want transaction stuck in mem pool because of low gas prices.

Reply:
Now, gas is estimated by chain,  we can take it into account later.